### PR TITLE
fix some errors #4670 

### PR DESCRIPTION
**Pull request by michalczemierowski**
### Purpose
#4670

### CODE REVIEW

File: UnityProject/Assets/Scripts/FluidPipe/PipeData.cs [hunk 1]
```diff
@@ -143,7 +145,12 @@ public virtual void OnDisable()
 					{
 						Outputs.Remove(Pipe);
 						//Removes itself if it is connected to a network and its a output for the network
-						Pipe.OnNet.RemoveEqualiseWith(this);
+
+						// this one probably require more work than just null check
+						if(Pipe.OnNet == null)
+							Logger.LogWarning("Pipe.OnNet == null", Category.Atmos);
+						else
+							Pipe.OnNet.RemoveEqualiseWith(this);
```
Comment (michalczemierowski) [inline UnityProject/Assets/Scripts/FluidPipe/PipeData.cs:153]: NullReferenceException: Object reference not set to an instance of an object
Pipes.PipeData.OnDisable () (at /github/workspace/UnityProject/Assets/Scripts/FluidPipe/PipeData.cs:146)
Pipes.MonoPipe.OnDisable () (at /github/workspace/UnityProject/Assets/Scripts/FluidPipe/MonoPipe.cs:33)

File: UnityProject/Assets/Scripts/FluidPipe/PipeFunctions.cs [hunk 2]
```diff
@@ -333,6 +333,8 @@ public void EqualiseWithOutputs(List<PipeData> others)
 			float density = this.Density();
 			foreach (var Pipe in others)
 			{
+				if(Pipe == null) continue;
+
```
Comment (michalczemierowski) [inline UnityProject/Assets/Scripts/FluidPipe/PipeFunctions.cs:337]: NullReferenceException: Object reference not set to an instance of an object
Pipes.PipeFunctions.PipeOrNet (Pipes.PipeData pipe) (at /github/workspace/UnityProject/Assets/Scripts/FluidPipe/PipeFunctions.cs:168)
Pipes.MixAndVolume.EqualiseWithOutputs (System.Collections.Generic.List`1[T] others) (at /github/workspace/UnityProject/Assets/Scripts/FluidPipe/PipeFunctions.cs:336)
Pipes.ReservoirAction.TickUpdate () (at /github/workspace/UnityProject/Assets/Scripts/FluidPipe/PipeActions.cs:47)
Pipes.PipeData.TickUpdate () (at /github/workspace/UnityProject/Assets/Scripts/FluidPipe/PipeData.cs:276)
AtmosManager.DoTick () (at /github/workspace/UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Atmospherics/AtmosManager.cs:86)
AtmosManager.Update () (at /github/workspace/UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Atmospherics/AtmosManager.cs:73)

File: UnityProject/Assets/Scripts/Messages/Server/SpritesMessages/SpriteUpdateMessage.cs [hunk 3]
```diff
@@ -203,11 +203,12 @@ public static void SendToAll(Dictionary<SpriteHandler, SpriteHandlerManager.Spri
...
 	{
 		foreach (var VARIABLE in ToSend)
-
 		{
+			if(VARIABLE.Value == null) continue;
+
 			ToReturn.Append(VARIABLE.Key.GetMasterNetID().netId.ToString());
 			ToReturn.Append("@");
-			ToReturn.Append(VARIABLE.Key.name);
+			ToReturn.Append(VARIABLE.Key?.name ?? "default_name");
```
Comment (michalczemierowski) [inline UnityProject/Assets/Scripts/Messages/Server/SpritesMessages/SpriteUpdateMessage.cs]: NullReferenceException
UnityEngine.Object.get_name () (at /home/builduser/buildslave/unity/build/Runtime/Export/Scripting/UnityEngineObject.bindings.cs:189)
SpriteUpdateMessage.GenerateStates (SpriteUpdateMessage spriteUpdateMessage, System.Collections.Generic.IEnumerable`1[T] ToSend) (at /github/workspace/UnityProject/Assets/Scripts/Messages/Server/SpritesMessages/SpriteUpdateMessage.cs:210)
SpriteUpdateMessage.GenerateMessage (System.Collections.Generic.IEnumerable`1[T] ToSend) (at /github/workspace/UnityProject/Assets/Scripts/Messages/Server/SpritesMessages/SpriteUpdateMessage.cs:197)
SpriteUpdateMessage.SendToAll (System.Collections.Generic.Dictionary`2[TKey,TValue] ToSend) (at /github/workspace/UnityProject/Assets/Scripts/Messages/Server/SpritesMessages/SpriteUpdateMessage.cs:187)
SpriteHandlerManager.UpdateClients () (at /github/workspace/UnityProject/Assets/Scripts/Sprite Handler/SpriteHandlerManager.cs:102)
SpriteHandlerManager.LateUpdate () (at /github/workspace/UnityProject/Assets/Scripts/Sprite Handler/SpriteHandlerManager.cs:88)

File: UnityProject/Assets/Scripts/NPC/AI/MobFlee.cs [hunk 4]
```diff
@@ -72,6 +72,15 @@ IEnumerator FindValidWayPoint(Vector2 oppositeDir)
...
 		}
+
+		// check fleeTarget again because it can be destroyed after waiting ^
+		if(!fleeTarget)
+		{
+			// set fleeTarget to itself if null
+			fleeTarget = transform;
+			oppositeDir = Vector2.zero;
+		}
+
```
Comment (michalczemierowski) [inline UnityProject/Assets/Scripts/NPC/AI/MobFlee.cs:83]: NullReferenceException
MobFlee+<FindValidWayPoint>d__14.MoveNext () (at /github/workspace/UnityProject/Assets/Scripts/NPC/AI/MobFlee.cs:114)
UnityEngine.SetupCoroutine.InvokeMoveNext (System.Collections.IEnumerator enumerator, System.IntPtr returnValueAddress) (at /home/builduser/buildslave/unity/build/Runtime/Export/Scripting/Coroutines.cs:17)

File: UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixInfo.cs [hunk 5]
```diff
@@ -94,7 +94,7 @@ public override string ToString()
 	{
 		return Equals(Invalid)
 			? "[Invalid matrix]"
-			: $"[({Id}){GameObject.name},offset={Offset},pivot={MatrixMove?.Pivot},state={MatrixMove?.ServerState},netId={NetID}]";
+			: $"[({Id}){GameObject?.name ?? "MatrixInfo"},offset={Offset},pivot={MatrixMove?.Pivot},state={MatrixMove?.ServerState},netId={NetID}]";
```
Comment (michalczemierowski) [inline UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixInfo.cs]: NullReferenceException: Object reference not set to an instance of an object
MatrixInfo.ToString () (at /github/workspace/UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixInfo.cs:95)
VVUIElementHandler.Serialise (System.Object InObject, System.Type TypeOf) (at /github/workspace/UnityProject/Assets/Scripts/Variable Viewer/BookViewer/ElementDisplay/VariableViewerUIFunctionHandler.cs:157)
VariableViewerNetworking.ProcessBook (Librarian+Book _book) (at /github/workspace/UnityProject/Assets/Scripts/Variable Viewer/VariableViewerNetworking.cs:309)
BookNetMessage.Send (Librarian+Book _book, UnityEngine.GameObject ToWho) (at /github/workspace/UnityProject/Assets/Scripts/Messages/Server/VariableViewer/BookNetMessage.cs:22)
VariableViewer.SendBookToClient (Librarian+Book Book, UnityEngine.GameObject ToWho) (at /github/workspace/UnityProject/Assets/Scripts/Variable Viewer/BookViewer/VariableViewer.cs:117)
VariableViewer.RequestSendBook (System.UInt64 BookID, UnityEngine.GameObject ByWho) (at /github/workspace/UnityProject/Assets/Scripts/Variable Viewer/BookViewer/VariableViewer.cs:216)
OpenBookIDNetMessage.ValidateAdmin () (at /github/workspace/UnityProject/Assets/Scripts/Messages/Client/VariableViewer/OpenBookIDNetMessage.cs:21)
OpenBookIDNetMessage.Process () (at /github/workspace/UnityProject/Assets/Scripts/Messages/Client/VariableViewer/OpenBookIDNetMessage.cs:14)
GameMessageBase.Process (Mirror.NetworkConnection sentBy) (at /github/workspace/UnityProject/Assets/Scripts/Messages/GameMessageBase.cs:25)
ClientMessage.Process (Mirror.NetworkConnection sentBy) (at /github/workspace/UnityProject/Assets/Scripts/Messages/Client/ClientMessage.cs:14)
GameMessageBase.PreProcess (Mirror.NetworkConnection sentBy, GameMessageBase b) (at /github/workspace/UnityProject/Assets/Scripts/Messages/GameMessageBase.cs:18)
Mirror.MessagePacker+<>c__DisplayClass6_0`2[T,C].<MessageHandler>b__0 (Mirror.NetworkConnection conn, Mirror.NetworkR

File: UnityProject/Assets/Scripts/Variable Viewer/VariableViewerNetworking.cs [hunk 6]
```diff
@@ -205,7 +205,7 @@ public int GetSize()
 			return sizeof(ulong)
 				+ sizeof(char) * Title.Length
 				+ sizeof(char) * BookClassname.Length
-				+ BindedPages.Sum(x => x.GetSize())
+				+ (BindedPages == null ? 0 : BindedPages.Sum(x => x.GetSize()))
```
Comment (michalczemierowski) [inline UnityProject/Assets/Scripts/Variable Viewer/VariableViewerNetworking.cs:208]: NullReferenceException: Object reference not set to an instance of an object
VariableViewerNetworking+NetFriendlySentence.GetSize () (at /github/workspace/UnityProject/Assets/Scripts/Variable Viewer/VariableViewerNetworking.cs:101)
VariableViewerNetworking+NetFriendlyPage+<>c.<GetSize>b__7_0 (VariableViewerNetworking+NetFriendlySentence x) (at /github/workspace/UnityProject/Assets/Scripts/Variable Viewer/VariableViewerNetworking.cs:174)
System.Linq.Enumerable.Sum[TSource] (System.Collections.Generic.IEnumerable`1[T] source, System.Func`2[T,TResult] selector) (at <351e49e2a5bf4fd6beabb458ce2255f3>:0)
VariableViewerNetworking+NetFriendlyPage.GetSize () (at /github/workspace/UnityProject/Assets/Scripts/Variable Viewer/VariableViewerNetworking.cs:170)
VariableViewerNetworking.ProcessBook (Librarian+Book _book) (at /github/workspace/UnityProject/Assets/Scripts/Variable Viewer/VariableViewerNetworking.cs:332)
BookNetMessage.Send (Librarian+Book _book, UnityEngine.GameObject ToWho) (at /github/workspace/UnityProject/Assets/Scripts/Messages/Server/VariableViewer/BookNetMessage.cs:22)
VariableViewer.SendBookToClient (Librarian+Book Book, UnityEngine.GameObject ToWho) (at /github/workspace/UnityProject/Assets/Scripts/Variable Viewer/BookViewer/VariableViewer.cs:117)
VariableViewer.RequestSendBook (System.UInt64 BookID, UnityEngine.GameObject ByWho) (at /github/workspace/UnityProject/Assets/Scripts/Variable Viewer/BookViewer/VariableViewer.cs:216)
OpenBookIDNetMessage.ValidateAdmin () (at /github/workspace/UnityProject/Assets/Scripts/Messages/Client/VariableViewer/OpenBookIDNetMessage.cs:21)
OpenBookIDNetMessage.Process () (at /github/workspace/UnityProject/Assets/Scripts/Messages/Client/VariableViewer/OpenBookIDNetMessage.cs:14)
GameMessageBase.Process (Mirror.NetworkConnection sentBy) (at /github/workspace/UnityProject/Assets/Scripts/Messages/GameMessageBase.cs:25)
ClientMessage.Process (Mirror.NetworkConnection sentBy) (at /

File: UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixInfo.cs (hunk 5 above)
Comment (corp-0) [reply UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixInfo.cs]: using ``?`` on a type derived from UnityEngine.Object should be avoided. 

null checking with GameObject and Matrix should be done using the regular ``== null`` expression.

File: UnityProject/Assets/Scripts/Messages/Server/SpritesMessages/SpriteUpdateMessage.cs (hunk 3 above)
Comment (corp-0) [reply UnityProject/Assets/Scripts/Messages/Server/SpritesMessages/SpriteUpdateMessage.cs]: Here too, it is unconvenient but this should use the regulat ``== null`` expression for null checking.

If you're using Rider, you can see it even shows a warning
image

File: UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixInfo.cs (hunk 5 above)
Comment (DooblyNoobly) [reply UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixInfo.cs]: corp-0 is there a reason for avoiding it?

Comment (corp-0) [reply UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixInfo.cs]: https://github.com/JetBrains/resharper-unity/wiki/Possible-unintended-bypass-of-lifetime-check-of-underlying-Unity-engine-object

**Pull request merged**